flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/watermarkstatus/StatusWatermarkValve.java

```diff
@@ -94,11 +103,12 @@ public void inputWatermark(Watermark watermark, int channelIndex, DataOutput<?>
             if (watermarkMillis > channelStatuses[channelIndex].watermark) {
                 channelStatuses[channelIndex].watermark = watermarkMillis;
 
-                // previously unaligned input channels are now aligned if its watermark has caught
-                // up
-                if (!channelStatuses[channelIndex].isWatermarkAligned
-                        && watermarkMillis >= lastOutputWatermark) {
-                    channelStatuses[channelIndex].isWatermarkAligned = true;
+                if (channelStatuses[channelIndex].isWatermarkAligned) {
+                    alignedChannelStatuses.adjustModifiedElement(channelStatuses[channelIndex]);
```

Comment:
It's better to introduce a method like `adjustAlignedChannelStatuses(...)` and add some comments for it. It can be helpful to make the code easier for understanding.